spec/std/yaml/from_yaml_spec.cr

```diff
@@ -0,0 +1,374 @@
+    values.each do |value|
+      expect_raises(YAML::ParseException) do
+        Nil.from_yaml %("#{value}")
+      end
+    end
+  end
+
+  it "should raise if YAML null value is quoted with single quotes" do
+    values.each do |value|
+      expect_raises(YAML::ParseException) do
+        Nil.from_yaml "'#{value}'"
+      end
+    end
+  end
+
+  it "should raise if not a null value" do
+    expect_raises(YAML::ParseException) do
+      Nil.from_yaml "hello"
+    end
+  end
+end
+
+describe "Bool.from_yaml" do
+  describe "true values" do
+    it "should serialize and deserialize" do
+      Bool.from_yaml(true.to_yaml).should eq true
+    end
+
+    values = {"true", "True", "TRUE", "on", "On", "ON", "y", "Y", "yes", "Yes", "YES"}
+
+    it "should return true if a truthy value" do
+      values.each do |value|
+        Bool.from_yaml(value).should eq true
+      end
+    end
+
+    it "should raise if a truthy value is quoted with double quotes" do
+      values.each do |value|
+        expect_raises(YAML::ParseException) do
+          Bool.from_yaml %("#{value}")
+        end
+      end
+    end
+
+    it "should raise if truthy is quoted with single quotes" do
+      values.each do |value|
+        expect_raises(YAML::ParseException) do
+          Bool.from_yaml "'#{value}'"
+        end
+      end
+    end
+  end
+
+  describe "false values" do
+    it "should serialize and deserialize" do
+      Bool.from_yaml(false.to_yaml).should eq false
+    end
+
+    values = {"false", "False", "FALSE", "off", "Off", "OFF", "n", "N", "no", "No", "NO"}
+
+    it "should return true if a falsey value" do
+      values.each do |value|
+        Bool.from_yaml(value).should eq false
+      end
+    end
+
+    it "should raise if a falsey value is quoted with double quotes" do
+      values.each do |value|
+        expect_raises(YAML::ParseException) do
+          Bool.from_yaml %("#{value}")
+        end
+      end
+    end
+
+    it "should raise if falsey is quoted with single quotes" do
+      values.each do |value|
+        expect_raises(YAML::ParseException) do
+          Bool.from_yaml "'#{value}'"
+        end
+      end
+    end
+  end
+
+  it "should raise if not a bool value" do
+    expect_raises(YAML::ParseException) do
+      Bool.from_yaml "hello"
+    end
+  end
+end
+
+[Int8, Int16, Int32, Int64, UInt8, UInt16, UInt32, UInt64].each do |type|
+  describe "#{type.name}.from_yaml" do
+    it "should serialize and deserialize" do
+      type.from_yaml(type.new("1").to_yaml).should eq type.new("1")
+    end
+
+    it "should parse a number into an #{type.name}" do
+      type.from_yaml("1").should eq type.new("1")
+    end
+
+    it "should raise if quoted with double quotes" do
+      expect_raises(YAML::ParseException) do
+        type.from_yaml %("1")
+      end
+    end
+
+    it "should raise if quoted with single quotes" do
+      expect_raises(YAML::ParseException) do
+        type.from_yaml "'1'"
```

Comment:
I think it would be more consistent to use `%()` string delimiters for the single quote examples as well.